### PR TITLE
 Convert examples/learning/helloworld.html to await/async 

### LINKED ISSUE #14121: Convert examples/learning/helloworld.html to await/async
**Please do not work on this issue unless you have applied to the outreachy project.**

The [helloworld.html](https://github.com/mozilla/pdf.js/blob/master/examples/learning/helloworld.html) example currently uses the older style of `promise.then` and callbacks. Convert it to use an `async` function and `await` style.

### PR DESCRIPTION
**Pull request by cathydev**
**Issue this pull request solves:**

closes #14121

### CODE REVIEW

File: examples/learning/helloworld.html [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (Snuffleupagus) [inline examples/learning/helloworld.html:1]: Unrelated change, please revert.

File: examples/learning/helloworld.html [hunk 2]
```diff
@@ -29,42 +30,42 @@ <h1>'Hello, world!' example</h1>
...
-    //
-    // Fetch the first page
-    //
-    pdf.getPage(1).then(function(page) {
-      var scale = 1.5;
-      var viewport = page.getViewport({ scale: scale, });
-      // Support HiDPI-screens.
-      var outputScale = window.devicePixelRatio || 1;
+  (async () => {
+  const pdf = await loadingTask;
```
Comment (Snuffleupagus) [inline examples/learning/helloworld.html:34]: Sorry, but this doesn't actually work!
The `loadingTask` is *not* a Promise, but rather a structure with a `promise`-property; see https://github.com/mozilla/pdf.js/blob/f6d9d91965a97c80b6f5b335a27cdb4d36b1b9e2/src/display/api.js#L527-L601

Hence the *correct* change would be `const pdfDocument = await loadingTask.promise;`.

Furthermore, given that the example cannot possibly work as-is: please make sure to (in this case manually) test the changes that you make to ensure that your code actually works.
Testing this example locally, after making changes, includes:
 - Running `gulp dist-install`.
 - Running `gulp server`.
 - Opening http://localhost:8888/examples/learning/helloworld.html in a browser and check that the example loads and works correctly without e.g. any errors printed in the console.

File: examples/learning/helloworld.html [hunk 3]
```diff
@@ -29,42 +30,42 @@ <h1>'Hello, world!' example</h1>
...
-        : null;
+  //
+  // Render PDF page into canvas context
+  //
+  var renderContext = {
+    canvasContext: context,
+    transform,
+    viewport,
+  };
+  page.render(renderContext);
```
Comment (Snuffleupagus) [inline examples/learning/helloworld.html:66]: Note that this *entire* block of code should be indented with *two* more spaces.

**Review by Snuffleupagus — CHANGES_REQUESTED** [review]
Besides the inline comments, please improve the *commit message* to say what you're actually changing since it's not entirely clear otherwise without looking at the code; e.g. "Convert examples/learning/helloworld.html to use async/await".

**Comment by catherinemds** [comment]
Thanks, I had a little problem with npm install and that was the reason why of the error but everything is working now

File: examples/learning/helloworld.html [hunk 4]
```diff
@@ -28,43 +28,44 @@ <h1>'Hello, world!' example</h1>
...
+    canvas.style.width = Math.floor(viewport.width) + "px";
+    canvas.style.height = Math.floor(viewport.height) + "px";
 
-      var transform = outputScale !== 1
-        ? [outputScale, 0, 0, outputScale, 0, 0]
-        : null;
+    const transform =
+      outputScale !== 1 
+      ? [outputScale, 0, 0, outputScale, 0, 0] 
+      : null;
```
Comment (Snuffleupagus) [inline examples/learning/helloworld.html:57]: As mentioned in https://github.com/mozilla/pdf.js/pull/14138#discussion_r729062437, please ensure that this reads as follows (and without any trailing spaces):
```javascript
const transform = outputScale !== 1
  ? [outputScale, 0, 0, outputScale, 0, 0]
  : null;
```

File: examples/learning/helloworld.html [hunk 5]
```diff
@@ -75,4 +76,4 @@ <h2>JavaScript code:</h2>
       document.getElementById('script').text;
 </script>
 </body>
-</html>
+</html>
```
Comment (Snuffleupagus) [inline examples/learning/helloworld.html:79]: Please revert this change, i.e. re-add the line-break at the end here.

File: package.json [hunk 6]
```diff
@@ -69,5 +69,8 @@
     "type": "git",
     "url": "git://github.com/mozilla/pdf.js.git"
   },
-  "license": "Apache-2.0"
+  "license": "Apache-2.0",
+  "dependencies": {
+    "pdfjs-dist": "file:build/dist"
+  }
```
Comment (Snuffleupagus) [inline package.json:75]: Please revert all changes to the `package-lock.json` and `package.json` files.

File: examples/learning/helloworld.html [hunk 7]
```diff
@@ -28,43 +28,44 @@ <h1>'Hello, world!' example</h1>
   //
   // Asynchronous download PDF
   //
-  var loadingTask = pdfjsLib.getDocument(url);
-  loadingTask.promise.then(function(pdf) {
+  const loadingTask = pdfjsLib.getDocument(url);
+  (async () => {
+    const pdf = await loadingTask.promise;;
```
Comment (Snuffleupagus) [inline examples/learning/helloworld.html:33]: One semi-colon too many here.

**Review by Snuffleupagus — CHANGES_REQUESTED** [review]
This looks better now, but there's still a few small things to fix here.
Also, in the commit message the same sentence is repeated more than once; please ensure that the final commit message reads only "Convert examples/learning/helloworld.html to await/async".

**Review by Snuffleupagus — APPROVED** [review]
r=me; thanks for the patch!
